python/mxnet/gluon/block.py

```diff
@@ -1119,8 +1135,8 @@ def _call_cached_op(self, *args):
                                  .format(fmt, self._in_format))
 
         args_without_none = [ele for ele in args if ele is not None]
-        cargs = [args_without_none[i] if is_arg else i.data()
-                 for is_arg, i in self._cached_op_args]
+        cargs = [args_without_none[i] if is_arg else i
+                 for is_arg, i, _ in self._cached_op_args]
```

Comment:
Would there be an issue now when passing arrays on a different context? Ie, initially using gpu(0) but later using gpu(1). 
Because self._cached_op_args now contains the .data() array directly instead of the parameter (which may manage arrays on multiple devices)

Reply:
https://github.com/apache/incubator-mxnet/blob/d9ec943194dc88942932cc2d9a988b2fc4ce69ee/python/mxnet/gluon/block.py#L1422-L1429